File: ShopEase/src/main/java/com/ecommerce/shopease/models/Product.java
package com.ecommerce.shopease.models;


import com.ecommerce.shopease.services.CartItemService;
import com.ecommerce.shopease.services.CategoryService;
import jakarta.persistence.*;
import lombok.Data;

import java.util.Set;

@Entity
@Data
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false)
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "specifications")
    private String specifications;

    @Column(name = "stock")
    private int stock;

    @Column(name = "cost")
    private int cost;

    @Column(name = "provider")
    private String provider;

    @Column(name = "description")
    private String description;

    @Column(name = "image")
    private String image;

    @ManyToOne
    @JoinColumn(name = "category_id")
    private Category category;

    @OneToMany(mappedBy = "product")
    private Set<CartItem> cartItems;
}

File: shopease-frontend/src/app/product/product.ts
export class Product {
    id: number;
    name: string;
    specifications: string;
    stock: number;
    provider: string;
    description: string;
    categoryId: number;
    cost: number;
    image: string
  
    public constructor(
      id: number,
      name: string,
      specifications: string,
      stock: number,
      provider: string,
      description: string,
      categoryId: number,
      cost: number,
      image: string
    ) {
      this.id = id;
      this.name = name;
      this.specifications = specifications;
      this.stock = stock;
      this.provider = provider;
      this.description = description;
      this.categoryId = categoryId;
      this.cost = cost;
      this.image = image;
    }
  }
  